### DISCUSSION TITLE
CMS Marketplace export does not return all entries

### DISCUSSION THREAD

**Discussion by dallas-fletchall** [mia-platform/community#433] [Q&A] [answered]
### What product are you having troubles with?

Marketplace

### What Console version are you using?

v12.2.1

### Description

How can I export all items from the marketplace from within the CMS?

I have tried using the export button shown below,

![image](https://github.com/mia-platform/community/assets/130166321/7375f9e6-2a21-4337-b10a-e05810faafb4)

with the expectations that I get the number of entries to match the data at the bottom of the page,

![image](https://github.com/mia-platform/community/assets/130166321/4f4dee3f-20ed-4791-aa83-3bb13b92c4bb)

### Actual Outcome

Less exported items than the number of elements at the bottom of the image above, which all seem to be "PUBLIC", and some that do not seem valid (in red).

![image](https://github.com/mia-platform/community/assets/130166321/7f0ca11a-1bb8-4925-b65a-99b78b00d0a5)

### Expected Outcome

A matching number of exported items compared to the elements at the bottom of the page, without filtering out non-public, and all being valid.

**Comment by fredmaggiowski** [comment]
Hi @dallas-fletchall, thanks for reaching out.

As you noticed, at the moment the export button is only exporting PUBLIC items; I think we can definitely improve the backoffice export integration to match your current visualization, at least with the viewed tab. 

<details>
<summary>Marketplace CMS Page tabs</summary>
<img width="333" alt="immagine" src="https://github.com/mia-platform/community/assets/7142570/22800233-f05b-4684-addd-d651270c6784">
</details>

Would that work for you?

As for the items marked in red it feels like there was some issues in processing an item with a base64 that may have contained some characters that functions as line delimiters for the xlsx format, does this make sense reviewing the items that have been split in multiple excel lines?

**Reply by dallas-fletchall** [reply]
I am just trying to export all. Any method that supports seems fine. I would prefer the JSON output, but ultimately would expect the same data formatted based on selection. I posted the `csv` output because it was easier to show the count.

I initially was trying to export using `miactl`, but I could not get `miactl marketplace list` to even list all items shown in the CMS. I was planning on looping over each item and running `miactl marketplace get --object-id {{id}}`.

**Reply by fredmaggiowski** [reply]
Yes the format does not have implications on the amount of exported data.

I assumed it was excel from the screenshot (looked like the macOS Numbers App); we'll look into a possible bug in csv export and try to improve the export functionality of the CMS.

As for the miactl export seems like a nice use case we could definitely explore and implement.

**Comment by dallas-fletchall** [comment, marked as answer]
Mia-Platform v13.0.2 still has issues with CSV export (same as above). The Excel export generates this message when trying to open.

![image](https://github.com/mia-platform/community/assets/130166321/23f22e9b-a013-454f-8652-2c11f6b10d8d)

The JSON export matches the public number of marketplace items.

**Reply by albertotessarotto** [reply]
Hello! Glad that the JSON export is working properly. Regarding the CSV export, we will try internally to replicate the problem and give you updates soon
